enclave/src/lib.rs

```diff
@@ -321,12 +322,16 @@ pub unsafe extern "C" fn sync_chain_relay(
             error!("Block verification failed. Error : {:?}", e);
             return sgx_status_t::SGX_ERROR_UNEXPECTED;
         }
-    }
 
-    let calls = match scan_blocks_for_relevant_xt(blocks, node_url) {
-        Ok(c) => c,
-        Err(e) => return e,
-    };
+        match scan_block_for_relevant_xt(&signed_block.block, node_url) {
+            Ok(c) => calls.extend(c.into_iter()),
+            Err(_) => error!("Error executing relevant extrinsics"),
+        };
+
+        if update_states(signed_block.block.header, node_url).is_err() {
```

Comment:
@brenzi As the runtime does perform state updates at the end of block execution I suppose it does make sense that we also update the state after we executed the relevant extrinsics?